src/app/doc_exporter.cpp

```diff
@@ -1228,6 +1231,36 @@ Doc* DocExporter::createEmptyTexture(const Samples& samples, base::task_token& t
   return document.release();
 }
 
+void DocExporter::adjustSamplesPixelFormat(app::DocExporter::Samples& samples,
+                                           doc::PixelFormat newPixelFormat) const
+{
+  for (auto& sample : samples) {
+    Image* image = sample.image().get();
+    if (!image || image->pixelFormat() == newPixelFormat)
+      continue;
+
+    ImageSpec spec(ColorMode(newPixelFormat), image->width(), image->height(), image->maskColor());
+    ImageRef convertedImg(Image::create(spec));
+    if (!convertedImg.get())
+      continue;
+
+    clear_image(convertedImg.get(), 0);
+    render::Dithering dithering;
+    Sprite* sprite = sample.sprite();
+    render::convert_pixel_format(image,
+                                 convertedImg.get(),
+                                 newPixelFormat,
+                                 dithering,
+                                 sprite ? sprite->rgbMap(0) : nullptr,
+                                 sprite ? sprite->palette(0) : nullptr,
+                                 sprite ? sprite->backgroundLayer() : nullptr,
+                                 0,
+                                 0,
+                                 nullptr);
+    sample.image() = std::move(convertedImg);
```

Comment:
It looks like we need a `Sample::setImage()` function, or even better: that we could add a `Sample::setPixelFormat()` function to the `Sample` class itself to delegate this work (and remove the `Sample::image()` function).

Reply:
You right! Thanks